modules/tinymce/src/core/main/ts/selection/MultiClickSelection.ts

```diff
@@ -1,44 +1,78 @@
-import { SugarElement } from '@ephox/sugar';
+import { Compare, PredicateFind, SugarElement } from '@ephox/sugar';
 
+import DomTreeWalker from '../api/dom/TreeWalker';
 import Editor from '../api/Editor';
-import * as CaretFinder from '../caret/CaretFinder';
+import { isCaretCandidate } from '../caret/CaretCandidate';
 import { CaretPosition } from '../caret/CaretPosition';
-import { isTextBlock } from '../dom/ElementType';
+import { isBlock } from '../dom/ElementType';
 import * as NodeType from '../dom/NodeType';
 import * as RangeNormalizer from './RangeNormalizer';
 
-const isTextBlockNode = (node: Node): node is Element => NodeType.isElement(node) && isTextBlock(SugarElement.fromDom(node));
+const isBr = NodeType.isBr;
+const isText = NodeType.isText;
+const isContentEditableFalse = (elm: SugarElement<Node>): boolean => NodeType.isContentEditableFalse(elm.dom);
+const isRoot = (rootNode: Node) => (elm: SugarElement<Node>): boolean => Compare.eq(SugarElement.fromDom(rootNode), elm);
 
-const normalizeSelection = (editor: Editor) => {
-  const rng = editor.selection.getRng();
-  const startPos = CaretPosition.fromRangeStart(rng);
-  const endPos = CaretPosition.fromRangeEnd(rng);
+const getClosestBlock = (node: Node, rootNode: Node): Node =>
```

Comment:
Maybe rename this to closest scope and also make sure that it covers the case of triple clicking in this content:
```
<p contenteditable="false">foo<span contenteditable="true">hello</span>bar</p>
```

Since triple click on hello should only select `hello` not `foo` and `bar`. At least that is how the Chrome does it today I guess it's debatable if it should select the parent noneditable block in that case.